app/services/flow/flow_state_machine.rb

```diff
@@ -24,7 +24,7 @@ def update
 
       increment_step_name_counts
       analytics.public_send(
-        flow.step_handler(step).analytics_submitted_event,
+        flow.step_handler_instance(step).analytics_submitted_event,
```

Comment:
i wonder if we should add a ticket to revisit this when the flow state machine is removed